e2e/atls/atls_test.go

```diff
@@ -0,0 +1,191 @@
+		}
+		idKeyHash := sha512.Sum384(idKeyBytes)
+		opt.ValidateOpts.TrustedIDKeyHashes = [][]byte{idKeyHash[:]}
+		return nil
+	}
+
+	testCases := map[string]struct {
+		manifestModifyFunc func(*manifest.ValidatorOptions) error
+		wantError          bool
+	}{
+		"default values": {
+			manifestModifyFunc: func(_ *manifest.ValidatorOptions) error {
+				return nil
+			},
+			wantError: false,
+		},
+		"platformInfo flip SMT to false": {
+			manifestModifyFunc: func(opt *manifest.ValidatorOptions) error {
+				if opt.ValidateOpts.PlatformInfo.SMTEnabled == false {
+					return fmt.Errorf("SMT must be disabled by default")
+				}
+				opt.ValidateOpts.PlatformInfo.SMTEnabled = false
+				return nil
+			},
+			wantError: true,
+		},
+		"idKeyDigestReplace flip Debug to true": {
+			manifestModifyFunc: func(opt *manifest.ValidatorOptions) error {
+				if opt.ValidateOpts.GuestPolicy.Debug == true {
+					return fmt.Errorf("Debug must be disabled by default")
+				}
+				// All fields in GuestPolicy are primitive types, so this copies whole struct.
+				guestPolicy := opt.ValidateOpts.GuestPolicy
+				guestPolicy.Debug = true
+				return idKeyDigestReplace([48]byte(opt.ValidateOpts.Measurement), guestPolicy, opt)
+			},
+			wantError: true,
+		},
+		"idKeyDigestReplace flip bits in measurement": {
+			manifestModifyFunc: func(opt *manifest.ValidatorOptions) error {
+				measurement := opt.ValidateOpts.Measurement
+				measurement[0] ^= 0xFF
+				return idKeyDigestReplace([48]byte(measurement), opt.ValidateOpts.GuestPolicy, opt)
+			},
+			wantError: true,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			require.NoError(t, ct.Kubeclient.WithForwardedPort(ctx, ct.Namespace, "port-forwarder-coordinator", "1313", func(addr string) error {
+				assert := require.New(t)
+				require := require.New(t)
+				// Go never uses a proxy for connections to localhost. To enable proxy tests, we
+				// replace localhost with 0.0.0.0, which can be used as localhost on Linux and BSD.
+				addr = strings.Replace(addr, "localhost", "0.0.0.0", 1)
+
+				kdsCache := memstore.New[string, []byte]()
+				kdsGetter := certcache.NewCachedHTTPSGetter(kdsCache, certcache.NeverGCTicker, logger.WithGroup("kds-getter"))
+				opts, err := manifestParsed.SNPValidateOpts(kdsGetter)
+				require.NoError(err, "getting SNP validate options")
+				var validators []atls.Validator
+				for i, opt := range opts {
+					opt.ValidateOpts.HostData = coordPolicyHashBytes
+					assert.NoError(tc.manifestModifyFunc(&opt))
+					name := fmt.Sprintf("snp-%d-%s", i, strings.TrimPrefix(opt.VerifyOpts.Product.Name.String(), "SEV_PRODUCT_"))
+					validators = append(validators, snp.NewValidator(opt.VerifyOpts, opt.ValidateOpts, logger.WithGroup("validator"), name))
+				}
+
+				dialer := dialer.New(atls.NoIssuer, validators, atls.NoMetrics, nil, logger)
+				conn, err := dialer.Dial(ctx, addr)
+				assert.NoError(err, "dialing coordinator")
```

Comment:
```suggestion
				require.NoError(err, "dialing coordinator")
```
It does not make sense to continue if dialing failed, does it?